cfpb/consumerfinance.gov
# Fix AnswerPage description auto-truncation when RTF does not come first

We started getting 500 errors on AnswerPages with empty `short_answer` fields and VideoPlayer blocks placed first in the `answer_content` StreamField.

This fixes that by cleaning up AnswerPage's description-setting code to piggyback on/override the method recently put in place for CFGOVPages as a whole.

It also makes the `truncate_preview` function a more clearly generic `truncate_by_words_and_chars` utility function. A future commit can pull that out and make use of it on the aforementioned CFGOVPage `get_meta_description` method.


## Additions

- new `get_meta_description` method on AnswerPages, overriding the stock CFGOVPage method to look at `search_description`, `short_answer`, and just the first `text` block in its `answer_content` StreamField


## Changes

- `truncate_preview` becomes `truncate_by_words_and_chars` with new kwargs to pass in the word and character limits


## How to test this PR

1. Pull branch
2. Iterate on an Ask CFPB AnswerPage with various combinations of:
   - With and without a short answer
   - Text block first in the answer content streamfield
   - Video block first in the answer content streamfield (make sure to use a legitimate YouTube video ID)
3. Ensure that in each scenario, no 500s appear and that the meta description output remains good


## Notes and todos

- [ ] Pull `truncate_by_words_and_chars` out into a utils file and apply truncation to CFGOVPage's meta description generation, as well. (Probably in a separate PR.)


## Checklist



  - PR titles are used to generate the change log in [releases](../../releases); good ones make that easier to scan.
  - Consider prefixing, e.g., "Mega Menu: fix layout bug", or "Docs: Update Docker installation instructions".
  - [This repo’s docs](https://cfpb.github.io/consumerfinance.gov/) (edit the files in the `/docs` folder) – for basic, close-to-the-code docs on working with this repo
  - CFGOV/platform wiki on GHE – for internal CFPB developer guidance
  - CFPB/hubcap wiki on GHE – for internal CFPB design and content guidance
